sched/scheduler/scheduler.go

```diff
@@ -65,24 +65,20 @@ func (s *scheduler) ScheduleJob(job sched.Job) error {
 
 		// Start Running Job
 		s.wg.Add(1)
-		go s.runJob(job, sagaState, nodes)
+		go func() {
+			defer s.wg.Done()
```

Comment:
What is the intended behavior for BlockUnitlAllJobsCompleted()? We currently allow jobs to start even after that function returns and I'm not sure that's useful.